src/Emulator/Peripherals/Peripherals/Mocks/HPSHostController.cs

```diff
@@ -366,6 +506,13 @@ private void IssueCommand(byte[] data)
             currentSlave.Write(data);
         }
 
+        private bool SocketConnected(Socket s)
+        {
+            bool dataAvailableForReading = s.Poll(1000, SelectMode.SelectRead);
+            bool availableBytesIsEmpty = (s.Available == 0);
+            return !(dataAvailableForReading && availableBytesIsEmpty);
```

Comment:
Is this condition correct? This will evaluate to `true` when `dataAvailableForReading == false` whis I believe is not the intention?

Reply:
Same as this [comment](https://github.com/renode/renode-infrastructure/pull/70/files#r959130179)

Reply:
Right, we'll receive `false` after `s.Poll` times out which just means that the connection is active but there is no data to read.